### PR TITLE
Update Readme

### PR DESCRIPTION
**Pull request by fbenz**
Changes:
* link to DZone article
* link to Sping Auto REST Docs
* list old documentation
* shorten feature list - you can always go to documentation to read more
* list license

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -3,10 +3,8 @@
 [![Build Status](https://travis-ci.org/ScaCap/spring-auto-restdocs.svg?branch=master)](https://travis-ci.org/ScaCap/spring-auto-restdocs)
 [![Maven Central](https://maven-badges.herokuapp.com/maven-central/capital.scalable/spring-auto-restdocs-core/badge.svg)](https://maven-badges.herokuapp.com/maven-central/capital.scalable/spring-auto-restdocs-core/)
 
-**Full documentation at https://scacap.github.io/spring-auto-restdocs**
-
-The aim of this extension to Spring REST Docs is to help you to write
-even less -- both code and documentation.
+The aim of this extension to [Spring REST Docs](https://projects.spring.io/spring-restdocs/)
+is to help you to write even less - both code and documentation.
```
Comment (jmisur) [inline README.md]: extension to.. is to.. help you to.. 
maybe rephrase

File: spring-auto-restdocs-docs/index.adoc [hunk 2]
```diff
@@ -27,11 +27,11 @@ Features:
 * Jackson visitor that gathers the whole JSON structure and includes Javadoc
 and constraint annotations on the fields. It works for both request and
 response bodies. In addition to the constraint documentation support that
-is already Spring REST Docs, we automatically include the constraint message
-in the documentation and also added support for documenting constraint groups.
+is already in Spring REST Docs, the constraint message is automatically included
+in the documentation. Furthermore, constraint groups can be documented automatically.
```
Comment (jmisur) [inline spring-auto-restdocs-docs/index.adoc]: Documentation also supports constraint groups.

**Comment by fbenz** [comment]
Improved the two sentences.
